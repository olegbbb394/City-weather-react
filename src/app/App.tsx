import React from 'react';
import {AppProviders} from "../providers/app-providers";
import {Routes, Route} from "react-router-dom";
import Home from "../components/home/home";
import SingleCard from "../components/single-card/single-card";

function App() {

    return (
        <>
            <AppProviders>
                <Routes>
                    <Route path='/' element={<Home/>}/>
                    <Route path='/card' element={<SingleCard/>}/>
                </Routes>
            </AppProviders>
        </>
    );
}

export default App;