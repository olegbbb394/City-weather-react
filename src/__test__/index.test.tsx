import {render} from "../test-utils";
import {BrowserRouter} from 'react-router-dom';
import App from '../app/App';

describe('should renders index', () => {

    it('should renders index', () => {
        const root = document.createElement('div');
        root.id = 'root';
        document.body.appendChild(root);

        const index = render(
            <BrowserRouter>
                <App/>
            </BrowserRouter>
        );

        expect(root).toBeInTheDocument();
        expect(index).toMatchSnapshot();
    });
});