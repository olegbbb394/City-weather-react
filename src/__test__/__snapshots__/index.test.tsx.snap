// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders index should renders index 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="root"
    />
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-h9p1cd-MuiContainer-root"
      >
        <header>
          <h1
            class="MuiTypography-root MuiTypography-h3 css-88618i-MuiTypography-root"
          >
            City weather
          </h1>
        </header>
        <section>
          <div
            class="MuiBox-root css-gqssyb"
          >
            <div
              class="MuiFormControl-root css-1jveyxb-MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSuccess MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-150d6dw-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                id="selectId"
              >
                City
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl  css-1k8trn2-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="labelId selectId"
                  class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="selectId"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      City
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <form
            autocomplete="off"
            class="MuiBox-root css-xn6ruc"
            novalidate=""
          />
        </section>
        <section>
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-1p81imu-MuiGrid2-root"
          />
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-h9p1cd-MuiContainer-root"
    >
      <header>
        <h1
          class="MuiTypography-root MuiTypography-h3 css-88618i-MuiTypography-root"
        >
          City weather
        </h1>
      </header>
      <section>
        <div
          class="MuiBox-root css-gqssyb"
        >
          <div
            class="MuiFormControl-root css-1jveyxb-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSuccess MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-150d6dw-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="selectId"
            >
              City
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSuccess MuiInputBase-formControl  css-1k8trn2-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="labelId selectId"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="selectId"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    City
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <form
          autocomplete="off"
          class="MuiBox-root css-xn6ruc"
          novalidate=""
        />
      </section>
      <section>
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-1p81imu-MuiGrid2-root"
        />
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
